grails-app/controllers/org/pih/warehouse/api/LocationApiController.groovy

```diff
@@ -242,7 +242,12 @@ class LocationApiController extends BaseDomainApiController {
         if (existingLocation.isZoneLocation() && Location.findAllByZone(existingLocation)) {
             throw new IllegalArgumentException("You cannot delete zone that is assigned to a bin location ${params.id}")
         }
-        locationService.deleteLocation(existingLocation)
+
+        try {
+            locationService.deleteLocation(existingLocation)
+        } catch (org.springframework.dao.DataIntegrityViolationException e) {
+            throw new Exception("${warehouse.message(code: 'default.not.deleted.with.reason.message', args: [warehouse.message(code: 'location.label', default: 'Location'), existingLocation.id, e.message])}")
+        }
```

Comment:
I know the reason you've done it, but I also don't feel comfortable with that change - we should in my opinion come up with better error handling (in the `ErrorsController`), because just throwing everywhere `try/catch` feels to me like a "lazy solution", that doesn't solve the real problem.

Could you please debug the error properties inside the `ErrorsController` and paste a screenshot here, so that maybe we could come up with some solution?
Dariusz's suggestion with getting to know the code of the exception would be perfect, but I don't remember if it's possible.

Maybe a good expirement would be to use the data services' delete method in the `deleteLocation` method instead of the domain one? How would the error look like then? (Both in the UI and caught in the `ErrorsController`)?

Reply:
> Could you please debug the error properties inside the ErrorsController and paste a screenshot here, so that maybe we could come up with some solution?

For me it doesn't make sense, I remember a situation in which I based on error properties (handling validation exceptions) and it worked only locally, it broke handling those errors on obdev3. But I can paste a screenshot for you 😀

![image](https://github.com/openboxes/openboxes/assets/83239466/4f8077bc-c479-49be-8b99-c22a643daa8b)


> Maybe a good expirement would be to use the data services' delete method in the deleteLocation method instead of the domain one? How would the error look like then? (Both in the UI and caught in the ErrorsController)?

The main reason for my change was to make the error more meaningful for users, even if it will be an error from data service it still wouldn't be something as I have at this moment:
 `Location ff8081818a83f2ac018a83f6e2bf0003 could not be deleted - could not execute statement; SQL [n/a]; constraint [null]; nested exception is org.hibernate.exception.ConstraintViolationException: could not execute statement`